project/cfmc30/README.rst

```diff
@@ -0,0 +1,124 @@
+FDCT: Fast Discrete Cosine Transformation Library
+=================
+
+Basic Information
+=================
+
+FDCT is a Python library built using C++, designed to compute Discrete Cosine
+Transformations (DCT) and invert discrete cosine transformation efficiently.
+By leveraging O(nlog(n)) complexity, FDCT improves performance through optimized
+parallelism and support for CPU SIMD(Single Instruction, Multiple Data) instructions.
+The library is suitable for large-scale data transformation tasks, especially in digital
+signal and image processing.
+
+Problem to Solve
+================
+
+In the field of digital signal processing, transformations such as the Discrete Fourier
+Transformation (DFT) and Discrete Cosine Transformation (DCT) are indispensable.
+These transformations decompose signals or images into their frequency components.
+While Fast Fourier Transformation (FFT) has optimized the DFT to an O(nlog(n)) complexity,
+an analogous efficient method for DCT is still a necessity for modern applications.
+
+The DCT is widely used in image compression(JPEG), audio compression(MP3), and other signal
+processing tasks. However, as data sizes grow, performing DCT on high-resolution images or
+large datasets becomes vomputationaly expensive.
+
+FDCT provides a fast DCT algorithm that brings down the time complexity to O(nlog(n)) using
+parallelism and SIMD to take advantage of modern multicore processors.
+
+
+Prospective Users
+=================
+
+FDCP will be valuable in fieds where DCT is used extensively:
+
+* Image Compression
+* Audio Compression
+* Video Processing
+* Signal Processing
+
+System Architecture
+===================
+
+FDCT is built as a Python extension using C++, and interacts with Python codebases
+through Numpy arrays as the primary data structure. The key architectural elements
+are:
+
+#. Input/Output
+
+   * FDCT accepts a 1D or 2D NumPy array as input, where each element represents a
+     pixel intensity (for images) or a signal value.
```

Comment:
Image uses 2D array and you will need to handle majoring. Which majoring you will support?  Please make it clear in your proposal.  A plan for the support should be included.

Please note that 2D array is required by image processing.  If you do not process image, you may or may not need to deal with 2D array.

Reply:
I plain to do both 1D and 2D array for audio signal and image.

Reply:
Please make it explicit (in the proposal) which majoring of the 2D arrays you will support.

Reply:
OK, I will explicitly emphasize the primary support for 2D arrays.

Reply:
I still don't see your plan for [majoring](https://yyc.solvcon.net/en/latest/nsd/04matrix/matrix.html#row-major-2d-array).  Please take a look at the notes and research online and update the proposal.